components/workouts/LogWorkout.jsx

```diff
@@ -0,0 +1,132 @@
+ * {
+ *    name: String,
+ *    color: String,
+ *    sets: [{ weight: String, reps: String }]
+ * }
+ *
+ */
+const LogWorkout = (props) => {
+  const { exercises, name } = props;
+
+  const initialExerciseState = parseExercises(exercises);
+
+  const [exerciseState, setExerciseState] = useState(initialExerciseState);
+
+  const curryUpdateReps = (exerciseIndex) => (setIndex) => (reps) => {
+    const newExercise = [...exerciseState];
+
+    newExercise[exerciseIndex].sets[setIndex].reps = reps;
+    setExerciseState(newExercise);
+  };
+
+  const curryUpdateWeight = (exerciseIndex) => (setIndex) => (weight) => {
+    const newExercise = [...exerciseState];
+
+    newExercise[exerciseIndex].sets[setIndex].weight = weight;
+    setExerciseState(newExercise);
+  };
+
+  const curryUpdateCompleted = (exerciseIndex) => (setIndex) => () => {
+    const newExercise = [...exerciseState];
+    const { completed } = newExercise[exerciseIndex].sets[setIndex];
+
+    newExercise[exerciseIndex].sets[setIndex].completed = !completed;
+    setExerciseState(newExercise);
+  };
+
+  const curryAddSet = (exerciseIndex) => () => {
+    const newExercise = [...exerciseState];
+    const { sets } = newExercise[exerciseIndex];
+
+    // Grab the last set for this exercise and mimic it
+    const lastSet = _.last(sets);
+    newExercise[exerciseIndex].sets.push({
+      weight: lastSet.weight,
+      prevWeight: lastSet.prevWeight,
+      reps: lastSet.reps,
+      prevReps: lastSet.prevReps,
+      completed: false,
+    });
+    setExerciseState(newExercise);
+  };
+
+  const curryDeleteSet = (exerciseIndex) => (setIndex) => () => {
+    const newExercise = [...exerciseState];
+
+    _.remove(newExercise[exerciseIndex].sets, (val, index) => index === setIndex);
+    setExerciseState(newExercise);
+  };
+
+  const renderExerciseDetail = ({ item, index }) => (
+    <ExerciseDetails
+      name={item.name}
+      items={item.sets}
+      color={item.color}
+      updateReps={curryUpdateReps(index)}
+      updateWeight={curryUpdateWeight(index)}
+      updateCompleted={curryUpdateCompleted(index)}
+      onSetAdd={curryAddSet(index)}
+      onSetDelete={curryDeleteSet(index)}
+    />
+  );
+
+  return (
+    <SafeAreaView style={{ height: '100%' }}>
+      <TitleText>{name}</TitleText>
+      <StyledFinishButton onPress={() => alert(`Top set has ${exerciseState[0].sets[0].weight} lbs by ${exerciseState[0].sets[0].reps} reps`)} />
+      <KeyboardAwareFlatList
+        style={{ height: '100%' }}
+        data={exerciseState}
+        renderItem={renderExerciseDetail}
+        keyExtractor={(item, index) => item.name + index}
+        keyboardOpeningTime={100}
+        // this is for a react navigation bug making it not smooth scroll
+        extraHeight={-64}
+      />
+    </SafeAreaView>
+  );
+};
+
+LogWorkout.propTypes = {
+  exercises: PropTypes.array.isRequired,
+  name: PropTypes.string.isRequired,
+};
+
```

Comment:
The app breaks if you delete all the sets, then try to add a set.

I dont think we should let the user delete a set if there is only 1 set left. 

Also we should start to think about if we want the user to be able to customize their workout during a workout, (i.e. The row machine is occupied, so they want to switch it out with pull ups)

Reply:
Ah good catch, yeah I agree we shouldn't let them delete it if there is one set left

Reply:
Also yeah we should allow them to switch out, I think we need to add something in the UI for this functionality, maybe the ellipses on the top right of each ExerciseDetails component? @shriyan44

Reply:
![IMG_3753](https://user-images.githubusercontent.com/42504462/98402043-78edfa80-201b-11eb-84fb-4e96fcee77d2.jpeg)